nextmove/src/main/java/no/difi/meldingsutveksling/nextmove/DpfConversationResource.java

```diff
@@ -1,15 +1,51 @@
 package no.difi.meldingsutveksling.nextmove;
 
 import lombok.Data;
+import lombok.NoArgsConstructor;
 
 import javax.persistence.DiscriminatorValue;
 import javax.persistence.Entity;
+import java.time.LocalDateTime;
+import java.util.Map;
+
+import static no.difi.meldingsutveksling.ServiceIdentifier.DPF;
 
 @Entity
 @DiscriminatorValue("DPF")
+@NoArgsConstructor
 @Data
 public class DpfConversationResource extends ConversationResource {
 
-    DpfConversationResource() {}
+    private DpfConversationResource(String conversationId, String senderId, String receiverId) {
+        super(conversationId, senderId, receiverId, DPF);
+    }
+
+    private DpfConversationResource(String conversationId,
+                                    String senderId,
+                                    String senderName,
+                                    String receiverId,
+                                    String receiverName,
+                                    LocalDateTime lastUpdate,
+                                    Map fileRefs,
+                                    Map customProperties) {
+        super(conversationId, senderId, senderName, receiverId, receiverName, DPF, lastUpdate, fileRefs, customProperties);
+    }
+
+    public static DpfConversationResource of(String conversationId, String senderId, String receiverId) {
+        return new DpfConversationResource(conversationId, senderId, receiverId);
+    }
+
+    public static DpfConversationResource of(ConversationResource cr) {
+        DpfConversationResource dpf = new DpfConversationResource(cr.getConversationId(),
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "dpf". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1488)